### PR TITLE
feat: Allow passing TypeScript function

### PR DESCRIPTION
**Pull request by 4513ECHO**
SSIA

### CODE REVIEW

**Comment by uga-rosa** [comment]
LGTM
Can I merge now?

**Comment by uga-rosa** [comment]
LGTM

File: denops/@ddc-sources/nvim-lsp.ts [hunk 1]
```diff
@@ -48,7 +48,9 @@ export type UserData = {
 };
 
 export type Params = {
-  snippetEngine: string; // ID of denops#callback. Required!
+  snippetEngine:
+    | string // ID of denops#callback. Required!
```
Comment (uga-rosa) [inline denops/@ddc-sources/nvim-lsp.ts:52]: This is my update omission, can you delete `Required!`?

Comment (4513ECHO) [reply denops/@ddc-sources/nvim-lsp.ts]: Done.

**Pull request merged**